platforms/core-execution/execution-e2e-tests/src/integTest/groovy/org/gradle/integtests/StaleOutputIntegrationTest.groovy

```diff
@@ -113,7 +113,7 @@ class StaleOutputIntegrationTest extends AbstractIntegrationSpec {
         succeeds(taskWithSources.taskPath)
 
         then:
-        overlappingOutputFile.exists()
+        overlappingOutputFile.parentFile.exists()
```

Comment:
❌ I don't think that is good. The overlapping file should not be deleted by the task that has empty sources. If it does get deleted from time to time then I'd call this a bug.